js/data/cryptoListingCurrencies.js

```diff
@@ -0,0 +1,167 @@
+import _ from 'underscore';
+import app from '../app';
+import fiatCurrencies from '../data/currencies';
+import {
+  getExchangeRates,
+  getEvents as getCurrencyEvents,
+} from '../utils/currency';
+
+// Since the exchange rate api is dependant on 3rd parties, we can't absolutely rely on
+// consistent results, so we'll start with a hard-coded list of crypto-listing currencies.
+// The base list could be updated from release to release. If there are additional crypto
+// currencies being returned by the exchange rate api, they will be merged into a combined
+// list the UI will use.
+const baseCurrencies = [
```

Comment:
The way this works is that the list of available coin types for crypto currency listings starts with a hard coded list. If the exchange rate api returns additional crypto currencies, they will be added to the list.

If a particular currency is in the base list but is not returned in the exchange rate api, it will still be available as a coin type and the UI will reflect that exchange rate data is not available and the listing won't be purchasable until that time. It may be that the exchange rate for that coin is temporarily unavailable. If it's a more permanent thing, then we can up the base list before the next release.

I thought this was the best way of having the app not be dependent on the exchange rate api for coin types (the api and/or one of the 3rd parties it pulls from may be down), yet be able to adjust to new coin types being available (i.e. if a new coin type is being returned by the exchange rate api, we don't need a new release for it to be available in the app).